plugin/impl.go

```diff
@@ -90,178 +81,52 @@ func (p *Plugin) Execute() error {
-						if err != nil {
-							if waiting {
-								continue
-							}
-							return fmt.Errorf("unable to trigger deploy for %s - err %v", entry, err)
-						}
-						fmt.Printf("starting deploy for %s/%s env - %s build - %d\n", owner, name, p.settings.Deploy, build.Number)
-						logParams(p.settings.params, p.settings.ParamsEnv.Value())
-
-						if p.settings.Block {
-							err = blockUntilBuildIsFinished(p, client, owner, name, int(newBuild.Number))
-							if err != nil {
-								return err
-							}
-						}
-
-						break I
-					}
-				}
+		fmt.Printf("successfully started Drone build for %s/%s: #%d\n", owner, name, build.ID)
+		logParams(p.settings.params, p.settings.ParamsEnv.Value())
 
-				// get the latest build for the specified repository
-				build, err := client.BuildLast(owner, name, branch)
-				if err != nil {
-					if waiting {
-						continue
-					}
-					return fmt.Errorf("unable to get latest build for %s: %s", entry, err)
-				}
-				if p.settings.Wait && !waiting && (build.Status == drone.StatusRunning || build.Status == drone.StatusPending) {
-					fmt.Printf("BuildLast for repository: %s, returned build number: %v with a status of %s. Will retry for %v.\n", entry, build.Number, build.Status, p.settings.Timeout)
-					waiting = true
-					continue
-				} else if p.settings.LastSuccessful && build.Status != drone.StatusPassing {
-					builds, err := client.BuildList(owner, name, drone.ListOptions{})
-					if err != nil {
-						return fmt.Errorf("unable to get build list for %s", entry)
-					}
-
-					build = nil
-					for _, b := range builds {
-						if b.Source == branch && b.Status == drone.StatusPassing {
-							build = b
-							break
-						}
-					}
-					if build == nil {
-						return fmt.Errorf("unable to get last successful build for %s", entry)
-					}
-				}
-
-				if (build.Status != drone.StatusRunning && build.Status != drone.StatusPending) || !p.settings.Wait {
-					// rebuild the latest build
-					var newBuild *drone.Build
-					newBuild, err = client.BuildRestart(owner, name, int(build.Number), p.settings.params)
-					if err != nil {
-						if waiting {
-							continue
-						}
-						return fmt.Errorf("unable to trigger build for %s", entry)
-					}
-					fmt.Printf("Restarting build %d for %s\n", build.Number, entry)
-					logParams(p.settings.params, p.settings.ParamsEnv.Value())
-
-					if p.settings.Block {
-						err = blockUntilBuildIsFinished(p, client, owner, name, int(newBuild.Number))
-						if err != nil {
-							return err
-						}
-					}
-
-					break I
-				}
+		if p.settings.Block {
+			if err := blockUntilBuildIsFinished(p, client, owner, name, int(build.Number)); err != nil {
+				return err
 			}
 		}
 	}
 
 	return nil
 }
 
-func parseRepoBranch(repo string) (string, string, string) {
-	var (
-		owner  string
-		name   string
-		branch string
-	)
-
+func parseRepoBranch(repo string) (string, string, string, error) {
```

Comment:
I don't have any strong feelings on the modification to this function but maybe others do @tboerger ?

Reply:
I think it's way better to return errors from this function at least, instead of leaving to the caller to surmise an error happened, without having the reason leading up to it.